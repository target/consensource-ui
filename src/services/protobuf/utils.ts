import {
  CertificateRegistryPayload,
  ICertificateRegistryPayload,
} from './compiled';

export interface PayloadInfo {
  payloadBytes: string | Buffer | NodeJS.TypedArray | DataView;
  inputs: string[];
  outputs: string[];
}

export const ACTIONS = CertificateRegistryPayload.Action;

/**
 * Encodes a CertificateRegistryPayload message
 * @param message CertificateRegistryPayload message or plain object to encode
 */
export function encodePayload(
  payload: ICertificateRegistryPayload,
): Uint8Array {
  return CertificateRegistryPayload.encode(payload).finish();
}
