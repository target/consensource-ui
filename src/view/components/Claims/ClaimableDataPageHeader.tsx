import React, { FC } from 'react';
import { Typography, Grid, makeStyles, createStyles } from '@material-ui/core';
import { ClaimedIconButton } from './ClaimedIconButton';
import { UnverifiedDataAlert } from './UnverifiedDataAlert';

const useStyles = makeStyles(
  createStyles({
    claimedIconBtn: {
      marginTop: 12.5,
      marginLeft: 7.5,
    },
  }),
);

export interface ClaimableDataPageHeaderProps {
  title: string;
  subtitle?: React.ReactNode;
  isClaimed: boolean;
}

export const ClaimableDataPageHeader: FC<ClaimableDataPageHeaderProps> = ({
  title,
  subtitle,
  isClaimed,
}) => {
  const classes = useStyles();

  return (
    <Grid container alignItems="center" direction="column" spacing={2}>
      <Grid item container justify="center" spacing={2}>
        <Grid item>
          <Typography variant="h2" align="center">
            {title}
          </Typography>
        </Grid>

        {isClaimed && (
          <Grid item>
            <div className={classes.claimedIconBtn}>
              <ClaimedIconButton fontSize="large" />
            </div>
          </Grid>
        )}
      </Grid>

      <Grid item>{subtitle}</Grid>

      {!isClaimed && (
        <Grid item xs={6}>
          <UnverifiedDataAlert />
        </Grid>
      )}
    </Grid>
  );
};
