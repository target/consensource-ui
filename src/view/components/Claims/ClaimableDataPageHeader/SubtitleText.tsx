import React, { FC } from 'react';
import { Typography } from '@material-ui/core';

/**
 * Formatted `<Typography>` used as the subtitle for
 * `<ClaimableDataPageHeader />`
 */
export const SubtitleText: FC = ({ children }) => {
  return (
    <Typography align="center" color="textSecondary">
      {children}
    </Typography>
  );
};
