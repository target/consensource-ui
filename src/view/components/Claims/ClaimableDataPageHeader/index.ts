export * from './ClaimableDataPageHeader';
export * from './SubtitleText';
