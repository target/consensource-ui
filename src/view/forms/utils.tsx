import React from 'react';
import { Grid, Typography, TextFieldProps } from '@material-ui/core';
import { BatchStatusRes } from 'services/api';

export interface TransactionFormProps {
  setBatchStatusLink: (statusLink: BatchStatusRes['link']) => void;
}

export type onChangeEvent = Parameters<
  NonNullable<TextFieldProps['onChange']>
>[0];

/**
 * Helper function that checks if any values in `state` have not been set.
 * Returns true if there is any key in `state` that is falsey.
 */
export function hasEmptyFields<T>(state: T) {
  return Object.keys(state).some((key) => !state[key as keyof T]);
}

export interface FormErrMsgProps {
  msg?: string;
}

export function FormErrMsg({ msg }: FormErrMsgProps) {
  if (!msg) {
    return null;
  }

  return (
    <Grid item>
      <Typography variant="caption" color="error">
        {msg}
      </Typography>
    </Grid>
  );
}
