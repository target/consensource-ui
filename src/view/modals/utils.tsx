import { DialogProps as MUIDialogProps } from '@material-ui/core';
import { FactoryResData } from 'services/api';

export interface DialogProps extends MUIDialogProps {
  handleClose: () => void;
}

export interface OrgDialogProps extends DialogProps {
  existing_org: FactoryResData;
}
