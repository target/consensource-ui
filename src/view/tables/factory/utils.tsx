import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
