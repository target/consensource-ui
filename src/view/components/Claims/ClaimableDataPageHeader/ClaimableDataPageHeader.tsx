import React, { FC } from 'react';
import { Typography, Grid, makeStyles, createStyles } from '@material-ui/core';
import { isDataClaimed } from 'utils';
import { SubtitleText } from './SubtitleText';
import {
  ClaimedIconButton,
  CLAIMED_ICON_BTN_WIDTH,
} from '../ClaimedIconButton';
import { UnverifiedDataAlert } from '../UnverifiedDataAlert';

const useStyles = makeStyles(
  createStyles({
    claimedIconBtn: {
      marginTop: 12.5,
      marginLeft: 7.5,
    },
  }),
);

export interface ClaimableDataPageHeaderProps {
  title: string;
  subtitle?: string | React.ReactNode;
  /**
   * Used to determine if the data is claimed or unclaimed
   */
  data: Record<string, any>;
}

export const ClaimableDataPageHeader: FC<ClaimableDataPageHeaderProps> = ({
  title,
  subtitle,
  data,
}) => {
  const classes = useStyles();
  const isClaimed = isDataClaimed(data);

  /**
   * Empty element used to balance out the `<ClaimedIconButton />`
   * for centering
   */
  const Balancer = () => (
    <Grid item style={{ width: CLAIMED_ICON_BTN_WIDTH }}>
      <div />
    </Grid>
  );

  return (
    <Grid container alignItems="center" direction="column" spacing={2}>
      <Grid item container justify="center" spacing={2}>
        {isClaimed && <Balancer />}

        <Grid item>
          <Typography variant="h2" align="center">
            {title}
          </Typography>
        </Grid>

        {isClaimed && (
          <Grid item>
            <div className={classes.claimedIconBtn}>
              <ClaimedIconButton fontSize="large" />
            </div>
          </Grid>
        )}
      </Grid>

      {subtitle && (
        <Grid item>
          {typeof subtitle === 'string' ? (
            <SubtitleText>{subtitle}</SubtitleText>
          ) : (
            subtitle
          )}
        </Grid>
      )}

      {!isClaimed && (
        <Grid item xs={6}>
          <UnverifiedDataAlert />
        </Grid>
      )}
    </Grid>
  );
};
