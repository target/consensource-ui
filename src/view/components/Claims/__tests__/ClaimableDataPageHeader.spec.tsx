import React from 'react';
import { render, screen } from 'utils/testing';
import { ClaimableDataPageHeader } from '../ClaimableDataPageHeader';

describe('<ClaimableDataPageHeader />', () => {
  it('renders with claimed data', () => {
    const { container } = render(
      <ClaimableDataPageHeader title="cert" isClaimed />,
    );

    expect(screen.getByLabelText('claimed-icon')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders with unclaimed data', () => {
    const { container } = render(
      <ClaimableDataPageHeader title="cert" isClaimed={false} />,
    );

    expect(screen.getByRole('alert')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders with a subtitle', () => {
    const { container } = render(
      <ClaimableDataPageHeader
        title="cert"
        isClaimed
        subtitle={<div data-testid="subtitle" />}
      />,
    );

    expect(screen.getByTestId('subtitle')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
