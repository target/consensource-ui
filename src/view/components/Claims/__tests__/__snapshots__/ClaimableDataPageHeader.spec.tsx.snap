// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClaimableDataPageHeader /> renders with a subtitle 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
        >
          cert
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="makeStyles-claimedIconBtn"
        >
          <div>
            <button
              aria-label="claimed-icon"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              title="Claimed Data"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm7 10c0 4.52-2.98 8.69-7 9.93-4.02-1.24-7-5.41-7-9.93V6.3l7-3.11 7 3.11V11zm-11.59.59L6 13l4 4 8-8-1.41-1.42L10 14.17z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        data-testid="subtitle"
      />
    </div>
  </div>
</div>
`;

exports[`<ClaimableDataPageHeader /> renders with claimed data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
        >
          cert
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="makeStyles-claimedIconBtn"
        >
          <div>
            <button
              aria-label="claimed-icon"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              title="Claimed Data"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm7 10c0 4.52-2.98 8.69-7 9.93-4.02-1.24-7-5.41-7-9.93V6.3l7-3.11 7 3.11V11zm-11.59.59L6 13l4 4 8-8-1.41-1.42L10 14.17z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
  </div>
</div>
`;

exports[`<ClaimableDataPageHeader /> renders with unclaimed data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
        >
          cert
        </h2>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    />
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div>
        <div
          class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
          role="alert"
        >
          <div
            class="MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message"
          >
            <div
              class="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Unverified Data
            </div>
            The source of this data is from third party sources
          </div>
          <div
            class="MuiAlert-action"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                LEARN MORE
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
